### DISCUSSION TITLE
(not) understanding what Dependency shows about a Component

### DISCUSSION THREAD

**Discussion by gurucubano** [DependencyTrack/dependency-track#1629] [Q&A]
If I look into a specific `component` in my project `SunRise/sisis-pap`,
for example the `component` httpd-2.4.53.tar.gz, this shows in the detailed
view:
```
Name: httpd-2.4.53.tar.gz
Version: 2.4.53
CPE: cpe:/a:apache:http_server:2.4.53
```
In the vulnerabilities tab it says 26 issues, and if I pick up this one:
```
CVE-2010-0434	5 Mar 2010	CWE-200 Exposure of Sensitive Information to an Unauthorized Actor
```
and go into it, it reads:
```
Overview

The ap_read_request function in server/protocol.c in the Apache HTTP Server 2.2.x before 2.2.15, when a multithreaded MPM is used, ...
```
i.e. it speaks about the versions before **2.2.15**!!!

I'm asking me (and you) why is this shown by DT? The version is **2.4.53** (the
most latest version, bleeding edge of Apache)? What might be wrong with our
data or with DT?

**Comment by stevespringett** [comment]
Well, the NVD has a ton of garbage data. CVE-2010-0434 specifically has `cpe:2.3:a:apache:http_server:-:*:*:*:*:*:*:*` which indicates that every version of Apache HTTP Server is affected. The JSON feed specifically lists that CPE as being vulnerable. 

The NVD needs to correct this erroneous data. Contact the NVD here: https://nvd.nist.gov/info

**Reply by stevespringett** [reply]
Actually the `-` in the version field for the CPE indicates that the version is not applicable. But DT will still make the same analysis on these.

**Reply by gurucubano** [reply]
Sticking with the case of CVE-2010-0434, do you mean because the NVD is listing here
https://nvd.nist.gov/vuln/detail/CVE-2010-0434#vulnConfigurationsArea
also a CPE with as dash in the version field:
```
 Known Affected Software Configurations Switch to CPE 2.2
 
 cpe:2.3:a:apache:http_server:-:*:*:*:*:*:*:*          <-----------------------------!!!
 cpe:2.3:a:apache:http_server:2.2:*:*:*:*:*:*:*
 cpe:2.3:a:apache:http_server:2.2.0:*:*:*:*:*:*:*
 cpe:2.3:a:apache:http_server:2.2.1:*:*:*:*:*:*:*
...
```
that this leads DT to showing this vulnerability also for the 2.4.53 version of the Apache? 

I think that would mean the DT interprets a `-` in the version field as `any` rather than `NA`, and isn't this rather a bug in DT then?

**Reply by officerNordberg** [reply]
@gurucubano I feel your frustration.  Personally I WANT DT to find this apache vulnerability because it could apply to SBOMs we import. From my experience I want DT to find more false positives than false negatives (within reason).  I think triaging for false positives is a sadly necessary function of vulnerability management given all or most of the datasources have their weaknesses. CPE matching and NVD data is frustratingly vague at times but FAR more complete than OSS Index. I currently have a gson vulnerability that is not being found by DT because the component doesn't have a CPE.   I see OSS Index has updates coming May 23rd.  I'm not holding my breath but have my fingers crossed

 I'm not sure if a global suppress feature that would allow organizations to increase their signal to noise ratio or a DT community built DB of aliases and refinement to the data could be possible.  Is there a way to submit CPE updates to NVD?

**Reply by gurucubano** [reply]
The component http-2.4.53 was noted with 26 vulnerabilities (7 high, 18 medium, 1 low); I went through the audit process and marked all as "False Positive" and "suppressed", which is a slow and painful process with DT. But, they remain presented in the colored analysis, and even in the component
overview where it says vulnerabilities=0 the colors remain, see picture
![http-2](https://user-images.githubusercontent.com/8157672/168971701-a24dffaa-6296-4258-9b93-483ce7f68b07.png)


At least a reload of the Project with the same BOM XML file does not reset the "False Positive" and "suppressed" markings.
